### PR TITLE
Add rule to verify Task bundle ref

### PR DESCRIPTION
**Pull request by lcarva**
https://issues.redhat.com/browse/HACBS-748

Signed-off-by: Luiz Carvalho <lucarval@redhat.com>

### CODE REVIEW

**Comment by lcarva** [comment]
Docs are tripping on this. Need to figure that out.

**Review by zregvart — COMMENTED** [review]
Those `_` methods are hard to follow, perhaps we should have some documentation and/or tests for them

**Comment by lcarva** [comment]
> Those `_` methods are hard to follow, perhaps we should have some documentation and/or tests for them

@zregvart, ack. I added docs to each one of them. Tests already existed for `_stream` and `_tag_by_digest`. I think the `_collection*` ones are better tested through the existing `warn` tests, but I can add duplicated tests if it helps. Let me know!

**Comment by lcarva** [comment]
@hacbs-contract/devs ptal
